backend/routes/users.js

```diff
@@ -9,31 +9,37 @@ const router = express.Router();
 router.post("/users/signup", async (req, res) => {
     const { username, password, email } = req.body;
 
-    try {
-        const userAlreadyExists = await User.findOne({
-            where: { [Op.or]: [{ username }, { email }] }
-        });
+    const userAlreadyExists = await User.findOne({
+        where: { [Op.or]: [{ username }, { email }] }
+    });
 
-        if (userAlreadyExists) {
-            return res.status(400).json({ error: "Username or email already exists." })
-        }
+    if (userAlreadyExists) {
+        return res.status(409).json({ error: "Username or email already exists." });
+    }
 
-        const hashedPassword = await bcrypt.hash(password, 10);
+    const hashedPassword = await bcrypt.hash(password, 10);
```

Comment:
I like you are doing the hashing on the back. If you do it on the front it can let hackers get the hash and get into the system, this kind of a attack is known as "pass the hash" https://www.beyondtrust.com/resources/glossary/pass-the-hash-pth-attack